### PR TITLE
chore: bump `near-sdk` to 5.18.1 

### LINKED ISSUE #1578: bump near-sdk to 5.18.1
### Background

near-sdk has a critical bug where the `private` modifier is ignored in combination with `init` attribute. The newly released version, `5.18.1`, has fixed this security bug.

### User Story

_No response_

### Acceptance Criteria

Upgrade near-sdk to 5.18.1 or latest version

### Resources & Additional Notes

_No response_

### PR DESCRIPTION
**Pull request by DSharifi**
closes #1578 

All changes, are due to breaking changes in the new sdk version:
- `.detach()` must be called for every promise, as the new version has #[must_use] attribute for `Promise` type, instead of silently dropping it.
- `ContractState` must be explicitly implemented for the contract state, `MpcContract`. It has a default implementation, but allows us to override it which is nice.

### CODE REVIEW

**Review by gilcu3 — DISMISSED** [review]
LGTM!

**Review by gilcu3 — APPROVED** [review]
